src/core/lib/channel/promise_based_filter.h

```diff
@@ -113,6 +121,9 @@ class ChannelFilter {
           [](const V3Based& v3) { return v3.instance_id; });
     }
 
+    const Blackboard* old_blackboard() const { return old_blackboard_; }
```

Comment:
Discussion point: whilst we need the old and the new blackboard held at this layer, I'm not sure we need to expose them out.

What if we have a method here -- `GetOrCreateBlackboardObject<T>` that first looks at the new blackboard, if it has a `T` then it returns it, if not, it looks at the old blackboard, if *it* has a `T` then we return that, otherwise it creates a new `T`, adds it to the new blackboard, and returns that.

My thinking here is that that API would cover the majority of uses of the blackboard system that *I* can predict, and by doing so we alleviate a decent amount of boilerplate whenever this use case comes up.

Reply:
I thought about possible ways to make this more seemless, but I think there could be cases where the filter might need its behavior to be a bit more nuanced.  For example, in the case of the RLQS filter, the blackboard entry will contain the RLQS client, which maintains a streaming call to a control plane and a bunch of associated state.  Normally, we'd want to retain that across filter updates, but there could be cases where we'd want to decide to recreate the state instead -- e.g., if we get an xDS update that changes the configuration to talk to a different RLQS server.  We *might* be able to handle those cases by simply making the properties that we want to retain be part of the key in the blackboard, but that might also be too simplistic -- we could see cases where the filter wants to retain only a strict subset of the state, in which case it might need to grab data from the entry in the old blackboard without actually retaining the entire entry.

I think we'll learn more about what we actually need here when we get into the RLQS implementation.  Until then, my preference would be to keep this a bit more general.

Reply:
I'd actually like to push back here and request doing the easy to type thing first, and if we need a more general mechanism - let's design that later when we've got the use case better scoped.

Reply:
Okay, done.

I will note that one down-side of this approach is that in the GCP auth filter, we wind up resetting the cache size immediately after initially creating the cache, which is a little weird.  But it didn't seem worth adding yet another lambda to the interface for code to be executed only when reusing an existing object.